### DISCUSSION TITLE
Localize alias

### DISCUSSION THREAD

**Discussion by cuni0716** [alexandercerutti/passkit-generator#193] [Q&A] [answered]
Hello @alexandercerutti and first of all thanks for this project, I'm using it since a few years ago and always worked like a charm.
Today I have a question that I don't know how to solve and maybe you knowledge will be the key to solve it.
I'm using the `localize` method in order to set the `pass.strings` files in the final bundle, I'm using the html code in order to convert a backfield value into a link, the href value for that link will change for every pass that I create and I want to localize the Link word.

Here is an example of what I'm doing:
```typescript
pkPass.backFields.push({ key: "rewards", label: "rewards.label", value: `<a href="${customerProfileUrl}">Link</a>` });
```

Any idea how can I achieve that?

Thanks in advance and thanks for the project!

**Comment by alexandercerutti** [comment]
Hi @cuni0716, thanks for having used passkit-generator for such long time ❤️ 

First of all, HTML is valid only when using `attributeValue` instead of `value`. `value` should be used to show the unstyled value where `attributeValue` is not supported (like in WatchOS). Look at it here: https://developer.apple.com/documentation/walletpasses/passfieldcontent

Second thing, I never tried to associate `attributeValue` with a localized string. I mean, for the documentation `attributeValue` supports l10n. I think that if you add the HTML to `pass.strings` and then add the translation reference you use in `pass.string` to `attributeValue`, it should link it correctly.

As I never tried you use case, I cannot ensure `pass.string` composition / reading will happen successfully, so if it doesn't work, I suggest you giving a look at the generated `pass.strings` file, to see if there is any problem.

Let me know if this works. I'm very curious about your use case!

**Reply by alexandercerutti** [reply]
Let me add a detail: if there is something that is not working with passkit-generator, and you can open the pkpass (as zip) to inspect the content, then you can try to follow the generating certificates guide: https://github.com/alexandercerutti/passkit-generator/wiki/Generating-Certificates#do-i-really-need-passkit-generator-to-create-apple-wallet-passes

This section will let you generate via CLI the `manifest.json` and the `signature`, in case if you want to change things manually and test them

**Reply by cuni0716** [reply]
The attributed value is not changed bu the value on the pass.strings file, so I think that this field does not support i18n, generated files looks correct by the way.

**Reply by alexandercerutti** [reply]
What's wierd is that the documentation says that the string is localizable, so it should take the value from `pass.strings`... I can try making a test this evening if needed

**Reply by cuni0716** [reply]
don't worry, just for you to know ;)

**Reply by alexandercerutti** [reply]
@cuni0716 I'm still curious ahah
Do you mind sending me the structure of your pass so I can imitate it?

**Reply by cuni0716** [reply]
Sure @alexandercerutti , just tell me what do you need. A sample pkpass file would be enough?

**Reply by alexandercerutti** [reply]
Yeah, sure!

**Reply by cuni0716** [reply]
here you have ;) (somehow github does not allow me to upload .pkpass file so I upload it in zip mode), hope it works!

Let me know if you found something and thanks again for the project and for your help!

[pass.zip](https://github.com/alexandercerutti/passkit-generator/files/14744261/pass.zip)

**Reply by alexandercerutti** [reply]
Sure, I'll let you know, thank you!

**Reply by alexandercerutti** [reply, marked as answer]
Hey there, @cuni0716, I tried to give a look and I successfully obtained what we talked about.

Look at this:

https://github.com/alexandercerutti/passkit-generator/assets/5442545/d727a63c-772f-4e94-a82d-d99063f093ea

The issue is that the html quotes must be escaped, as the pass format requires string to be surrounded by double quotes. Something like this must appear in the final file:

```strings
"rewards.html" = "<a href=\"https://github.com/alexandercerutti/passkit-generator\">View all rewards</a>";
```


I don't know if, in both cases, single quotes work.

I'm not sure if there's anything that should be fixed in `passkit-generator`, but in your code you just need to double-escape the double quotes, like `\\"`, as the first will be interpreted by JS I guess.

Also. I think that `%@` or whatever only works within the `changeMessage` field.

Let me know if it works!

**Reply by cuni0716** [reply]
Ohh yes, the problem was there, by using typescript I've needed to add 3 scape slahes like this:
`"rewards.anchor": "<a href=\\\"<LINK>\\\">View all rewards</a>",`
but it worked, thanks a lot again @alexandercerutti !!

**Reply by alexandercerutti** [reply]
You are welcome @cuni0716!
